tests/kafkatest/services/kafka.py

```diff
@@ -199,29 +208,53 @@ def restart_node(self, node, wait_sec=0, clean_shutdown=True):
     def leader(self, topic, partition=0):
         """ Get the leader replica for the given topic and partition.
         """
-        cmd = "/opt/kafka/bin/kafka-run-class.sh kafka.tools.ZooKeeperMainWrapper -server %s " \
-              % self.zk.connect_setting()
-        cmd += "get /brokers/topics/%s/partitions/%d/state" % (topic, partition)
-        self.logger.debug(cmd)
+        topic_partition_data = self.zk_get_data("/brokers/topics/%s/partitions/%d/state" % (topic, partition))
```

Comment:
These paths don't account for ZK chroots. Or is that just handled by the zkconnect setting?